addons/sale_project/controllers/portal.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from operator import itemgetter
+
+from odoo import _
+from odoo.http import request
+from odoo.tools import groupby as groupbyelem
+
+from odoo.osv.expression import OR
+
+from odoo.addons.project.controllers.portal import ProjectCustomerPortal
+
+
+class SaleProjectCustomerPortal(ProjectCustomerPortal):
+
+    def _task_get_searchbar_groupby(self):
+        values = super()._task_get_searchbar_groupby()
+        values['sale_order'] = {'input': 'sale_order', 'label': _('Sales Order'), 'order': 4}
+        values['sale_line'] = {'input': 'sale_line', 'label': _('Sales Order Item'), 'order': 5}
+        return dict(sorted(values.items(), key=lambda item: item[1]["order"]))
+
+    def _task_get_groupby_mapping(self):
+        groupby_mapping = super()._task_get_groupby_mapping()
+        groupby_mapping.update(
+            sale_order='sale_order_id',
+            sale_line='sale_line_id')
+        return groupby_mapping
+
+    def _task_get_order(self, order, groupby):
+        if groupby == 'sale_order':
+            order = "sale_order_id, %s" % order
+        elif groupby == 'sale_line':
+            order = "sale_line_id, %s" % order
+        else:
+            order = super()._task_get_order(order, groupby)
+        return order
```

Comment:
If we have the suggestion that I propose in the parent method of this one, then we just need to override the `_task_get_groupby_mapping` as you did above this method and we can remove this override because the parent method will do the job thanks to the `_task_get_groupby_mapping` method.  😉